Framework/Algorithms/src/ResizeRectangularDetector.cpp

```diff
@@ -97,19 +97,19 @@ void ResizeRectangularDetector::exec() {
     throw std::runtime_error("Component with name " + ComponentName +
                              " was not found.");
 
-  auto &componentInfo = inputW->mutableComponentInfo();
-  const auto componentIndex = componentInfo.indexOf(comp->getComponentID());
-  if (!componentInfo.isDetector(componentIndex))
+  RectangularDetector_const_sptr det =
+      boost::dynamic_pointer_cast<const RectangularDetector>(comp);
```

Comment:
Given backwards compatibility for cases like this, and some others in `makeLegacyParameterMap` it would be nice to have some way of `ComponentInfo::isRectangularDetectorPixel`. That way we could avoid the dynamic cast and we are visiting each of these RectangularDetectorPixels anyway in the construction. We would only need a `std::vector<bool>` to store that information.